pkg/webui/console/components/device-map/index.js

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import MapWidget from '../../../components/map/widget/'
+
+export default class DeviceMap extends React.Component {
+  render() {
+    const { device } = this.props
+
+    const markers =
+      device.locations && device.locations.user
+        ? [
+            {
+              position: {
+                latitude: device.locations.user.latitude || 0,
+                longitude: device.locations.user.longitude || 0,
+              },
+            },
+          ]
+        : []
+
+    return (
+      <MapWidget
+        id="device-map-widget"
+        markers={markers}
+        path={`/applications/${device.ids.device_id}/devices/${device.ids.application_ids.application_id}/location`}
```

Comment:
Please destructs the `device` object  and extract the ids to make this shorted

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/1129/commits/b09b516e732a1f6e20d5b5e888eb021c56af8b5d

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/pull/1129/commits/cd0ea62e7e00c95a3718313cecce61fe5699da91